### ISSUE TITLE
Curly braces wrongly emitted in async functions when targeting ES5 or ES3

### ISSUE THREAD

**Issue by samvv** [Microsoft/TypeScript#8621]
When targeting ES5 or ES3, async functions will be generated without curly braces, throwing a syntax error in NodeJS. It would be nicer if a warning was emitted that tells the user that async functions are not supported for ES5 and ES3.

**TypeScript Version:** 

 - 1.8.10
 - 1.9.0-dev.20160428-1.0

**Code**

```typescript
let test = async () =>  {
  // optional
}
```

**Expected behavior:**

```javascript
Warning: async functions are not supported when targeting ES3 or ES5
```

**Actual behavior:** 

```javascript
var test = function () __awaiter(this, void 0, void 0, function* () {
});
```

```
/Users/samvv/Projects/node/test/test.js:9
var test = function () __awaiter(this, void 0, void 0, function* () {
                       ^^^^^^^^^

SyntaxError: Unexpected identifier
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:387:25)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:148:18)
    at node.js:405:3
```

**Comment by mhegazy** [comment]
why not use `--target ES6`?

**Comment by DanielRosenwasser** [comment]
The issue here is that `tsc` doesn't give an error when you have an async arrow function and you target ES5.

In contrast, if you target ES5 with an async function declaration, it will give an error message.

**Comment by samvv** [comment]
@mhegazy 
> why not use --target ES6?

Yes, indeed, it was a silly mistake of me. I didn't come to the realization that ES6 generators are, of course, needed. I've changed the description accordingly.

@DanielRosenwasser 
> The issue here is that tsc doesn't give an error when you have an async arrow function and you target ES5.
> 
> In contrast, if you target ES5 with an async function declaration, it will give an error message.
Ah, I see.

**Comment by mhegazy** [comment]
this should be handled by moving to the transforms pipeline. do not think there is a need to fix it in the current emitter.
